internal/cmd/marketplace/apply/apply.go

```diff
@@ -135,47 +192,45 @@ func buildFilePathsList(paths []string) ([]string, error) {
 	return filePaths, nil
 }
 
-func buildApplyRequest(pathList []string) (*marketplace.ApplyRequest, error) {
-	resources := []marketplace.Item{}
+func buildApplyRequest(pathList []string) (*marketplace.ApplyRequest, map[string]string, error) {
+	resources := []*marketplace.Item{}
 	resNameToFilePath := map[string]string{}
 	for _, filePath := range pathList {
 		content, err := os.ReadFile(filePath)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		var fileEncoding string
 		switch filepath.Ext(filePath) {
-		case encoding.YamlExtension, encoding.YmlExtension:
+		case encoding.YamlExtension, encoding.YmlExtension, encoding.JSONExtension:
 			fileEncoding = encoding.YAML
```

Comment:
If we know that the behavior of the json/yaml unmarshalling is not as expecte we may need to directly change the implementation of encoding package.
This code seems like a smell